tests/cecil-tests/AttributeTest.cs

```diff
@@ -0,0 +1,257 @@
+			foreach (var type in Helper.FilterTypes (assembly, a => HasAnyAvailabilityAttribute (a, includeUnsupported: true))) {
+				CheckCurrentPlatformIncludedIfAny (type, platformName, type.FullName, type.DeclaringType, found);
+			}
+			foreach (var prop in Helper.FilterProperties (assembly, a => HasAnyAvailabilityAttribute (a, includeUnsupported: true))) {
+				CheckCurrentPlatformIncludedIfAny (prop, platformName, prop.FullName, prop.DeclaringType, found);
+			}
+			foreach (var meth in Helper.FilterMethods (assembly, a => HasAnyAvailabilityAttribute (a, includeUnsupported: true))) {
+				CheckCurrentPlatformIncludedIfAny (meth, platformName, meth.FullName, meth.DeclaringType, found);
+			}
+
+			Assert.That (found, Is.Empty);
+		}
+
+		void CheckCurrentPlatformIncludedIfAny (ICustomAttributeProvider item, string platformName, string fullName, TypeDefinition parent, HashSet<string> found)
+		{
+			if (HasAnyAvailabilityAttribute (item, true)) {
+				var supportedAttributes = item.CustomAttributes.Where (a => IsSupportedAttribute (a));
+				if (!supportedAttributes.Any (a => FindAvailabilityKind (a) != platformName)) {
+					found.Add (fullName);
+				}
+			}
+		}
+
+		string AssemblyToAttributeName (string assemblyPath)
+		{
+			switch (Path.GetFileName (assemblyPath)) {
+			case "Xamarin.iOS.dll":
+				return "ios";
+			case "Xamarin.TVOS.dll":
+				return "tvos";
+			case "Xamarin.Mac.dll":
+				return "macos";
+			case "Xamarin.MacCatalyst.dll":
+				return "maccatalyst";
+			default:
+				throw new NotImplementedException ();
+			}
+		}
+
+		[Conditional ("DEBUG")]
+		void DebugPrint (string fullName, IEnumerable<string> parentAvailability, IEnumerable<string> myAvailability)
+		{
+			Console.WriteLine (fullName);
+			Console.WriteLine ("Parent: " + string.Join (" ", parentAvailability));
+			Console.WriteLine ("Mine: " + string.Join (" ", myAvailability));
+			Console.WriteLine ();
+		}
+
+		bool Ignore (string fullName)
+		{
+			switch (fullName) {
+			default:
+				return false;
+			}
+		}
+
+		bool FirstContainsAllOfSecond<T> (IEnumerable<T> first, IEnumerable<T> second)
+		{
+			var firstSet = new HashSet<T> (first);
+			return second.All (s => firstSet.Contains (s));
+		}
+
+		IEnumerable<string> GetAvailabilityAttributes (ICustomAttributeProvider provider, bool includeUnsupported) => GetAvailabilityAttributes (provider.CustomAttributes, includeUnsupported);
+
+		IEnumerable<string> GetAvailabilityAttributes (IEnumerable<CustomAttribute> attributes, bool includeUnsupported)
+		{
+			var availability = new List<string> ();
+			foreach (var attribute in attributes.Where (a => IsAvailabilityAttribute (a, includeUnsupported))) {
+				var kind = FindAvailabilityKind (attribute);
+				if (kind is not null) {
+					availability.Add (kind);
+				}
+			}
+			return availability;
+		}
+
+		// Unfortunate state we need to keep since I can't see to walk "up" from a
+		// MethodDefinition get_Foo or SetFoo to see it's container's properties
```

Comment:
Something like this should work:

```cs
var getter = ...;
var property = getter.DeclaringType.Properties.Find (v => v.Name == getter.Name.Substring (4));
```

Reply:
It seems to work, at least enough for this PR. I'll fix it up with the attribute changes in follow up.